### PR TITLE
SWC-5848

### PR DESCRIPTION
**Pull request by nickgros**
[Jira](https://sagebionetworks.jira.com/browse/SWC-5848)

We shouldn't show the Column editor (step 2) in the wizard for datasets.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/TableType.java [hunk 1]
```diff
@@ -13,13 +13,28 @@
...
+	projects(PROJECT, DisplayConstants.PROJECT_VIEW),
+	tables(TABLE, DisplayConstants.VIEW),
+	folders(FOLDER, DisplayConstants.VIEW),
+	folders_tables(FOLDER | TABLE, DisplayConstants.VIEW),
+	files(FILE, DisplayConstants.FILE_VIEW),
+	files_tables(FILE | TABLE, DisplayConstants.VIEW),
+	files_folders(FILE | FOLDER, DisplayConstants.VIEW),
+	files_folders_tables(FILE | FOLDER | TABLE, DisplayConstants.VIEW),
+	dataset(FILE, DisplayConstants.DATASET);
+
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/TableType.java:31]: Trying to come up with a reliable way to map each table type to what we typically call it in the UI.

In short:
TableEntity -> "Table"
SubmissionView -> "Submission View"
Dataset -> "Dataset"
EntityView with only files in mask -> "File View"
EntityView with only projects in mask -> "Project View"
EntityView with any other mask -> "View"

Did I get this right?

Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/TableType.java:31]: LGTM

File: src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/CreateTableViewWizardStep1.java [hunk 2]
```diff
@@ -134,8 +135,14 @@ private void createEntity(final Entity entity) {
 		jsClient.createEntity(entity).addCallback(new FutureCallback<Entity>() {
 			@Override
 			public void onSuccess(Entity table) {
-				step2.configure((Table) table, tableType);
-				modalPresenter.setNextActivePage(step2);
+				// For Datasets, this is the only step.
+				if (TableType.dataset.equals(tableType)) {
+					modalPresenter.onFinished();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/CreateTableViewWizardStep1.java:140]: Test in CreateTableViewWizardStep1Test to verify early exit for Datasets?

Comment (nickgros) [reply src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/CreateTableViewWizardStep1.java:140]: https://github.com/Sage-Bionetworks/SynapseWebClient/pull/4709/files#diff-f0702dba294bd57c898b4bd1693895c233db7ed5bb5063042687fd5651298f22R217

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**